src/Worker.Extensions.DurableTask/HTTP/HttpRetryOptions.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.DurableTask.Http;
+
+/// <summary>
+/// Defines retry policies that can be passed as parameters to various operations.
+/// </summary>
+public class HttpRetryOptions
+{
+    // Would like to make this durability provider specific, but since this is a developer
+    // facing type, that is difficult.
+    private static readonly TimeSpan DefaultMaxRetryinterval = TimeSpan.FromDays(6);
+
+    private TimeSpan firstRetryInterval;
+    private int maxNumberOfAttempts;
+    private IList<HttpStatusCode>? statusCodesToRetry;
+
+    /// <summary>
+    /// Creates a new instance SerializableRetryOptions with the supplied first retry and max attempts.
+    /// </summary>
+    /// <param name="firstRetryInterval">Timespan to wait for the first retry.</param>
+    /// <param name="maxNumberOfAttempts">Max number of attempts to retry.</param>
+    /// <param name="statusCodesToRetry">List of status codes that specify when to retry.</param>
+    /// <exception cref="ArgumentException">
+    /// The <paramref name="firstRetryInterval"/> value must be greater than <see cref="TimeSpan.Zero"/>.
+    /// </exception>
+    public HttpRetryOptions(TimeSpan firstRetryInterval, int maxNumberOfAttempts, IList<HttpStatusCode>? statusCodesToRetry = null)
+    {
+        this.MaxRetryInterval = DefaultMaxRetryinterval;
+
+        this.firstRetryInterval = firstRetryInterval;
+        this.maxNumberOfAttempts = maxNumberOfAttempts;
+        this.statusCodesToRetry = statusCodesToRetry ?? new List<HttpStatusCode>();
+    }
+
+    /// <summary>
+    /// Gets or sets the first retry interval.
+    /// </summary>
+    /// <value>
+    /// The TimeSpan to wait for the first retries.
+    /// </value>
+    [JsonPropertyName("FirstRetryInterval")]
+    public TimeSpan FirstRetryInterval
+    {
+        get { return this.firstRetryInterval; }
+        set { this.firstRetryInterval = value; }
+    }
+
+    /// <summary>
+    /// Gets or sets the max retry interval.
+    /// </summary>
+    /// <value>
+    /// The TimeSpan of the max retry interval, defaults to 6 days.
+    /// </value>
+    [JsonPropertyName("MaxRetryInterval")]
+    public TimeSpan MaxRetryInterval { get; set; }
```

Comment:
nit: instead of setting in the ctor, you can inline the default like so:

```suggestion
    public TimeSpan MaxRetryInterval { get; set; } = DefaultMaxRetryinterval;
```